src/VisualStudio/Core/Def/Implementation/TaskList/HostDiagnosticUpdateSource.cs

```diff
@@ -16,32 +16,32 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.TaskList
     [Export(typeof(HostDiagnosticUpdateSource))]
     internal sealed class HostDiagnosticUpdateSource : AbstractHostDiagnosticUpdateSource
     {
-        private readonly VisualStudioWorkspaceImpl _workspace;
+        private readonly Lazy<VisualStudioWorkspaceImpl> _workspace;
 
         private readonly object _gate = new object();
         private readonly Dictionary<ProjectId, HashSet<object>> _diagnosticMap = new Dictionary<ProjectId, HashSet<object>>();
 
         [ImportingConstructor]
-        public HostDiagnosticUpdateSource(VisualStudioWorkspaceImpl workspace, IDiagnosticUpdateSourceRegistrationService registrationService)
+        public HostDiagnosticUpdateSource(Lazy<VisualStudioWorkspaceImpl> workspace, IDiagnosticUpdateSourceRegistrationService registrationService)
```

Comment:
Can this just import VisualStudioWorkspace instead of "Impl"? "Impl" usually means "this is doing something evil with the guts of it" but I'm not sure I'm actually seeing that here.